module/assets/helper.js

```diff
@@ -0,0 +1,1103 @@
+            let typeOfAssetsToMove = reqData.assetInformation.objectType;
+            if (
+              reqData.toUserProfile.roles.includes(
+                constants.common.CONTENT_CREATOR
+              ) &&
+              typeOfAssetsToMove === constants.common.SOULTION
+            ) {
+              //match and Set Queries for solutions partial Transfer
+              let solutionFilter = {
+                _id: new ObjectId(reqData.assetInformation.identifier),
+                author: reqData.fromUserProfile.userId,
+              };
+              let updateSolutions = {
+                $set: {
+                  author: reqData.toUserProfile.userId,
+                  creator: reqData.toUserProfile.firstName,
+                },
+              };
+              let solutionLicenseFilter = {
+                author: reqData.fromUserProfile.userId,
+                license: { $exists: true, $ne: "" },
+              };
+              let updateSolutionsLicense = {
+                $set: {
+                  "license.author": reqData.toUserProfile.firstName,
+                  "license.creator": reqData.toUserProfile.firstName,
+                },
+              };
+
+              let updateUserSolutions = [
+                solutionsHelper.updateMany(solutionFilter, updateSolutions),
+                solutionsHelper.updateMany(
+                  solutionLicenseFilter,
+                  updateSolutionsLicense
+                ),
+              ];
+              let updatedOneToOneTransferSolution = await Promise.all(
+                updateUserSolutions
+              );
+              updatedOneToOneTransferSolution
+                ? (updateUserAssetDataResult = true)
+                : (updateUserAssetDataResult = false);
+            }
+            if (
+              (reqData.toUserProfile.roles.includes(
+                constants.common.PROGRAM_MANAGER
+              ) ||
+                reqData.toUserProfile.roles.includes(
+                  constants.common.PROGRAM_DESIGNER
+                )) &&
+              typeOfAssetsToMove === constants.common.PROGRAM
+            ) {
+              //get from and to user details from userExtension collection
+
+              let usersExtesionData = await userExtensionsHelper.find({
+                userId: {
+                  $in: [
+                    reqData.fromUserProfile.userId,
+                    reqData.toUserProfile.userId,
+                  ],
+                },
+              });
+              let fromUserData = usersExtesionData.find(
+                (fromUser) => fromUser.userId === reqData.fromUserProfile.userId
+              );
+              let toUserData = usersExtesionData.find(
+                (toUser) => toUser.userId === reqData.toUserProfile.userId
+              );
+              if (toUserData) {
+                //return promise after updating programs in userExtension collection
+                /**
+                   * @fromUserData
```

Comment:
@remove or add as comment ideally in between the data changes you don't need to add